File: pinyougou-sellergoods-interface/src/main/java/com/itheima/sellergoods/service/BrandService.java
package com.itheima.sellergoods.service;

import java.util.List;

import com.pinyougou.pojo.TbBrand;

import entity.PageResult;
import entity.Result;

/**
 * 品牌Service层接口
 * @author Jason
 *
 */

public interface BrandService {
	/**
	 * 查询所有品牌
	 * @return
	 */
	List<TbBrand> findAll();
	
	/**
	 * 分页查询
	 * @param pageNum	当前页
	 * @param pageSize	每页记录数
	 * @return
	 */
	PageResult findPage(Integer pageNum,Integer pageSize);
	
	/**
	 * 新增品牌
	 * @param brand
	 */
	void add(TbBrand brand);
	
	/**
	 * 查询品牌详情
	 * @param id
	 * @return
	 */
	TbBrand findOne(Long id);
	
	/**
	 * 修改品牌
	 * @param brand
	 */
	void update(TbBrand tbBrand);
	
	/**
	 * 删除品牌
	 * @param ids
	 */
	void delete(Long[] ids);
	
	/**
	 * 分页查询（可带条件）
	 * @param brand	查询条件
	 * @param pageNum	当前页
	 * @param pageSize	每页显示的数据
	 * @return
	 */
	PageResult findPageByCondition(TbBrand brand,Integer pageNum,Integer pageSize);

}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
/*
    		服务层 
    	*/
    	app.service("brandService", function($http) {
    		//查询所有品牌
    		this.findAll=function(){
    			return $http.get("/brand/findAll.do");
    		};
    		
    		//分页查询
    		this.findPage=function(pageNum,pageSize){
    			return $http.get("/brand/findPage.do?pageNum="+pageNum+"&pageSize="+pageSize);
    		};
    		
    		//新增品牌
    		this.add=function(entity){
    			return $http.post("/brand/add.do",entity)
    		};
    		
    		//修改品牌
    		this.update=function(entity){
    			return $http.post("/brand/update.do",entity)
    		};
    		//修改品牌回显品牌信息
    		this.findOne=function(id){
    			return $http.get("/brand/findOne.do?id="+id)
    		};
    		//删除品牌
    		this.dele=function(selectIds){
    			return $http.get("/brand/delete.do?ids="+selectIds)
    		};
    		//分页条件查询
    		this.search=function(pageNum,pageSize,searchEntity){
    			return $http.post("/brand/search.do?pageNum="+pageNum+"&pageSize="+pageSize,searchEntity)
    		}
    		//获取品牌列表
    		this.selectOptionList=function(){
    			return $http.get("/brand/selectOptionList.do");
    		}
    	});